massa-bootstrap/src/server.rs

```diff
@@ -305,12 +279,66 @@ impl BootstrapServer {
                 }
             }
         }
-        // wait for bootstrap sessions to finish
-        while bootstrap_sessions.next().await.is_some() {}
 
         Ok(())
     }
 
+    /// Helper method to check IP's bootstrap attempt history, consuming the server in the process.
+    ///
+    /// On success, will return the server for later use.
+    /// On failure, includes a side effect of sending an error message to the client.
+    ///
+    /// The intended use of the `Result` is to signal a `continue` to the calling context on failure.
+    /// ...I know, not the best. This function is largely a cut-paste of code that existed in the main-loop `BootstrapServer::run()` method, and where that code would invoke a `continue`, here we return `Err`
```

Comment:
Try to avoid using "I" in public comments

Reply:
Done